src/app.js

```diff
@@ -1,21 +1,36 @@
 'use strict';
 
-/**
- * Implement sum function:
- *
- * Function takes 2 numbers and returns their sum
- *
- * sum(1, 2) === 3
- * sum(1, 11) === 12
- *
- * @param {number} a
- * @param {number} b
- *
- * @return {number}
- */
-function sum(a, b) {
-  // write code here
-  return a + b;
-}
+const http = require('http');
 
-module.exports = sum;
+const server = http.createServer((req, res) => {
+  res.writeHead(200, { 'Content-Type': 'application/json' });
+
+  const normalizedURL = new URL(req.url, `http://${req.headers.host}`);
+  const path = new URL(req.url, `http://${req.headers.host}`).pathname.slice(1);
+  const parts = path.split('/');
+  const query = normalizedURL.search.slice(1);
+  const queryArrayPairs = query.split('&');
+  const queryObject = {};
+
+  for (const pair of queryArrayPairs) {
+    const pairArray = pair.split('=');
+    const key = pairArray[0];
+    const value = pairArray[1];
+    const existingValue = queryObject[key];
+
+    if (existingValue) {
+      queryObject[key] = typeof existingValue === 'object'
+        ? [...existingValue, value]
+        : [existingValue, value];
+    } else {
+      queryObject[key] = value;
+    }
+  }
```

Comment:
It's ok, but you can do it in another way)
```suggestion
  const url = req.url.split('?');
  const [parts, query] = url;
  const params = new URLSearchParams(query);

  const data = {
    parts: parts.slice(1).split('/'),
    query: Object.fromEntries(params),
  };
```